source/cgal/surface_mesh.cc

```diff
@@ -114,6 +116,105 @@ namespace CGALWrappers
                     (f % 2 == 0));
   }
 
+
+
+  template <typename CGALPointType, int dim, int spacedim>
+  void
+  boolean_operation(
+    const typename Triangulation<dim, spacedim>::cell_iterator &cell1,
+    const typename Triangulation<dim, spacedim>::cell_iterator &cell2,
+    CGAL::Surface_mesh<CGALPointType> &                         outsm,
+    const BooleanOperation &boolean_operation)
+  {
+    Assert(dim != 1 || spacedim != 1,
+           ExcMessage(
+             "This function does not work with 1-dimensional objects."));
+    Assert(
+      outsm.is_empty(),
+      ExcMessage(
+        "The output surface mesh needs to be empty upon calling this function."));
+    namespace PMP = CGAL::Polygon_mesh_processing;
+    CGAL::Surface_mesh<CGALPointType> sm1, sm2;
+    dealii_cell_to_cgal_surface_mesh(cell1,
+                                     MappingQ<dim, spacedim>(1),
+                                     sm1); //[TODO: remove Mapping from here...]
+    dealii_cell_to_cgal_surface_mesh(cell2, MappingQ<dim, spacedim>(1), sm2);
+    PMP::triangulate_faces(sm1);
+    PMP::triangulate_faces(sm2);
+
+    [[maybe_unused]] bool res = false;
+    switch (boolean_operation)
+      {
+        case BooleanOperation::UNION:
+          res = PMP::corefine_and_compute_union(sm1, sm2, outsm);
+          break;
+        case BooleanOperation::INTERSECTION:
+          res = PMP::corefine_and_compute_intersection(sm1, sm2, outsm);
+          break;
+        case BooleanOperation::DIFFERENCE:
+          res = PMP::corefine_and_compute_difference(sm1, sm2, outsm);
+          break;
+        case BooleanOperation::NONE:
```

Comment:
This is not going to produce anything visible on the outside. outsm is not going to be touched by this function, and sm1 and sm2 will be destroied at the end of the call. We should document what will happen in this case, i.e., we return sm1, i.e., the corefinement of the first cell, therefore here you need to do `outsm = std::move(sm1);`